### PR TITLE
Add ability to reset ACL bootstrap process

### PR DESCRIPTION
**Pull request by armon**
This PR allows the ACL bootstrapping process to be reset. The user creates a special file in `<datadir>/server/acl-bootstrap-reset` with the reset index, and then uses the standard ACL bootstrapping endpoint.

### CODE REVIEW

File: website/source/guides/acl.html.markdown [hunk 1]
```diff
@@ -104,7 +104,7 @@ $ curl \
 }
 ```
 
-Once the initial bootstrap is performed, it _cannot be performed again_. Make sure to save this AccessorID and SecretID.
+Once the initial bootstrap is performed, it cannot be performed again until [reset](#reseting-acl-bootstrap). Make sure to save this AccessorID and SecretID.
```
Comment (dadgar) [inline website/source/guides/acl.html.markdown]: until -> unless

File: nomad/acl_endpoint.go [hunk 2]
```diff
@@ -324,6 +345,32 @@ func (a *ACL) Bootstrap(args *structs.ACLTokenBootstrapRequest, reply *structs.A
...
+		if !os.IsNotExist(err) {
+			a.srv.logger.Printf("[ERR] acl.bootstrap: failed to read %q: %v", path, err)
+		}
+		return 0
+	}
+
+	// Attempt to parse the file
+	var resetIdx uint64
+	if _, err := fmt.Sscanf(string(raw), "%d", &resetIdx); err != nil {
+		a.srv.logger.Printf("[ERR] acl.bootstrap: failed to parse %q: %v", path, err)
```
Comment (dadgar) [inline nomad/acl_endpoint.go]: Why wouldn't you return this error to the user?

File: nomad/fsm_test.go [hunk 3]
```diff
@@ -1623,6 +1623,27 @@ func TestFSM_BootstrapACLTokens(t *testing.T) {
 	out, err := fsm.State().ACLTokenByAccessorID(nil, token.AccessorID)
 	assert.Nil(t, err)
 	assert.NotNil(t, out)
+
+	// Teset with reset
```
Comment (dadgar) [inline nomad/fsm_test.go]: teset

**Comment by dadgar** [comment]
Build error: https://travis-ci.org/hashicorp/nomad/builds/273982621#L1011

File: nomad/acl_endpoint.go (hunk 2 above)
Comment (armon) [reply nomad/acl_endpoint.go]: Seems odd to get an API error because of a file existing... It's a pretty edge case scenario either way.

**Pull request merged**